src/playlist.js

```diff
@@ -36,9 +36,27 @@
     // accumulate the segment durations into the result
     for (; i < endSequence; i++) {
       segment = playlist.segments[i - playlist.mediaSequence];
-      result += segment.preciseDuration ||
```

Comment:
Yep. `preciseDuration` didn't account for time-shifted audio and video tracks or the gaps between PTS values across multiple segments. That could add up to tens of milliseconds of error per segment.